geth/account/accounts_test.go

```diff
@@ -103,9 +107,373 @@ func TestVerifyAccountPasswordWithAccountBeforeEIP55(t *testing.T) {
+	err := s.accManager.SelectAccount(s.address, s.password)
+	s.NoError(err)
+
+	testCases := []struct {
+		name                  string
+		address               string
+		password              string
+		accountKeyStoreReturn []interface{}
+		expectedError         error
+	}{
+		{
+			"success",
+			s.address,
+			s.password,
+			[]interface{}{s.keyStore, nil},
+			nil,
+		},
+		{
+			"fail_keyStore",
+			s.address,
+			s.password,
+			[]interface{}{nil, testErrKeyStore},
+			testErrKeyStore,
+		},
+		{
+			"fail_wrongAddress",
+			"wrong-address",
+			s.password,
+			[]interface{}{s.keyStore, nil},
+			ErrAddressToAccountMappingFailure,
+		},
+		{
+			"fail_wrongPassword",
+			s.address,
+			"wrong-password",
+			[]interface{}{s.keyStore, nil},
+			errors.New("cannot retrieve a valid key for a given account: could not decrypt key with given passphrase"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		s.T().Run(testCase.name, func(t *testing.T) {
+			s.reinitMock()
+			s.nodeManager.EXPECT().AccountKeyStore().Return(testCase.accountKeyStoreReturn...).AnyTimes()
+			childAddr, childPubKey, err := s.accManager.CreateChildAccount(testCase.address, testCase.password)
+			if testCase.expectedError != nil {
+				s.Equal(testCase.expectedError, err)
+			} else {
+				s.NoError(err)
+				s.NotEmpty(childAddr)
+				s.NotEmpty(childPubKey)
+			}
+		})
+	}
+}
+
+func (s *ManagerTestSuite) TestSelectedAndReSelectAccount() {
+	// Select the test account
+	s.nodeManager.EXPECT().AccountKeyStore().Return(s.keyStore, nil).AnyTimes()
+	s.nodeManager.EXPECT().WhisperService().Return(s.shh, nil).AnyTimes()
+	err := s.accManager.SelectAccount(s.address, s.password)
+	s.NoError(err)
+
+	s.T().Run("success", func(t *testing.T) {
+		acc, err := s.accManager.SelectedAccount()
+		s.NoError(err)
+		s.NotNil(acc)
+
+		err = s.accManager.ReSelectAccount()
+		s.NoError(err)
+	})
+
+	s.T().Run("ReSelect_fail_whisper", func(t *testing.T) {
+		s.reinitMock()
+		s.nodeManager.EXPECT().WhisperService().Return(nil, testErrWhisper).AnyTimes()
+		err = s.accManager.ReSelectAccount()
+		s.Equal(testErrWhisper, err)
+	})
+
+	s.accManager.selectedAccount = nil
+	s.reinitMock()
+	s.nodeManager.EXPECT().AccountKeyStore().Return(s.keyStore, nil).AnyTimes()
+	s.nodeManager.EXPECT().WhisperService().Return(s.shh, nil).AnyTimes()
+
+	s.T().Run("Selected_fail_noAccount", func(t *testing.T) {
+		_, err := s.accManager.SelectedAccount()
+		s.Equal(ErrNoAccountSelected, err)
+	})
+
+	s.T().Run("ReSelect_success_noAccount", func(t *testing.T) {
+		err = s.accManager.ReSelectAccount()
+		s.NoError(err)
+	})
+}
+
+func (s *ManagerTestSuite) TestLogout() {
+	s.nodeManager.EXPECT().WhisperService().Return(s.shh, nil)
+	err := s.accManager.Logout()
+	s.NoError(err)
+
+	s.nodeManager.EXPECT().WhisperService().Return(nil, testErrWhisper)
+	err = s.accManager.Logout()
+	s.Equal(testErrWhisper, err)
+}
+
+// TestAccounts tests cases for (*Manager).Accounts
```

Comment:
Top 👍 . Only interpunction missing. Yeah, I know, sometimes pedantic. 😁

Reply:
Do you mean like `Manager.Accounts` instead?

Reply:
Oh, eh, no. Just

```
// TestAccounts tests cases for (*Manager).Accounts.
```

I do like full sentences for comments, also no friend `// do this and that` (w/o interpunction and starting with a lower-case letter in case of an inline comment). As I said, I'm personally a bit pedantic. 🤣

Reply:
Ah, okay 😆 Adding the dot, no worries. lol